File: test/corpus/languages/typescript/fixture/src/main.test.ts
import { createSampleUser } from "./main";
import { UserRepository, MemoryStorage } from "./user";

function testCreateSampleUser() {
    const user = createSampleUser();
    console.assert(user.name === "John");
}

function testUserRepository() {
    const storage = new MemoryStorage();
    const repo = new UserRepository(storage);
    const user = createSampleUser();
    repo.addUser(user);
    console.assert(repo.countUsers() === 1);
}

File: test/corpus/languages/go/fixture/main_test.go
package main

import "testing"

func TestCreateSampleUser(t *testing.T) {
	user := createSampleUser()
	if user.Name != "John" {
		t.Errorf("expected John, got %s", user.Name)
	}
}

func TestUserRepository(t *testing.T) {
	storage := NewMemoryStorage()
	repo := NewUserRepository(storage)
	user := createSampleUser()
	err := repo.AddUser(user)
	if err != nil {
		t.Fatal(err)
	}
}
